### PR TITLE
Implement various trivial methods required by GitUp

### PR DESCRIPTION
**Pull request by ethanc8**
There are a bunch of methods and enums that are needed by GitUp (and probably a lot of other modern apps) that I have implemented in these commits. I think some issues appeared while rebasing the commits on master, though, so these commits might need to be changed.

Additionally, this adds a bunch of renamed enums (Apple has renamed a lot of enums to allow the Swift bridging generator to automatically parse the enum names).

### CODE REVIEW

File: Headers/AppKit/NSView.h
Comment (fredkiefer) [inline Headers/AppKit/NSView.h:43]: You probably don't need both the import and the class declaration.

File: Headers/AppKit/NSAlert.h
Comment (fredkiefer) [inline Headers/AppKit/NSAlert.h:60]: Shouldn't we reuse the type and block type you define in NSWindow.h here?

File: Headers/AppKit/NSApplication.h
Comment (fredkiefer) [inline Headers/AppKit/NSApplication.h:79]: Looks like this is the third place where you define this type.

File: Source/NSApplication.m
Comment (fredkiefer) [inline Source/NSApplication.m:3847]: We prefer:
`if (_appearance)
{
   return _appearance;
}`

File: Source/NSView.m
Comment (fredkiefer) [inline Source/NSView.m:5168]: This looks like an error. You seem to have removed all the layout code that was added last year.

**Review by fredkiefer — CHANGES_REQUESTED** [review]
Thank you for this great addition. What worries me is that the layout changes seem to have been lost. Could you please have a look at that?

**Comment by ethanc8** [comment]
@fredkiefer I believe I've implemented the changes you've requested. Is there anything else I need to do?

**Review by fredkiefer — APPROVED** [review]
Looks great now!

**Comment by gcasa** [comment]
There are some issues with your changes regarding @property in NSAppearance.  Please correct these before merging.

**Comment by ethanc8** [comment]
> There are some issues with your changes regarding `@property` in NSAppearance. Please correct these before merging.

@gcasa I see that I've forgotten to implement these properties. I might have accidentally removed that when I rebased on the current master. Are there any issues in the header file?

**Comment by ethanc8** [comment]
[From the GCC CI runner](https://github.com/gnustep/libs-gui/actions/runs/7791417046/job/21260786141?pr=240):
```
In file included from ../Headers/AppKit/NSView.h:37,
                 from Functions.m:44:
../Headers/AppKit/NSAppearance.h:89:11: error: unknown property attribute ‘strong’
   89 | @property(strong) NSAppearance* appearance;
      |           ^~~~~~
../Headers/AppKit/NSAppearance.h:89:33: warning: object property ‘appearance’ has no ‘assign’, ‘retain’ or ‘copy’ attribute; assuming ‘assign’
   89 | @property(strong) NSAppearance* appearance;
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
../Headers/AppKit/NSAppearance.h:89:33: note: ‘assign’ can be unsafe for Objective-C objects; please state explicitly if you need it
../Headers/AppKit/NSAppearance.h:93:21: error: unknown property attribute ‘strong’
   93 | @property(readonly, strong) NSAppearance* effectiveAppearance;
      |                     ^~~~~~
```
@gcasa Is there a macro that shows whether `@property(strong)` is supported by the compiler, or should I just detect clang?

**Comment by ethanc8** [comment]
> Is there a macro that shows whether `@property(strong)` is supported by the compiler, or should I just detect clang?
[The clang docs](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#property-declarations) imply that `@property(retain)` and `@property(strong)` are equivalent, and GCC supports `@property(retain)`, so I'll just use that. I've added some notes about it [to my Objective-C documentation project](https://ethanc8.github.io/NewDocumentation-ObjectiveC/Reference/Properties.html#property-attributes) ([source](https://github.com/ethanc8/NewDocumentation-ObjectiveC/blob/master/source/Reference/Properties.md))

**Comment by ethanc8** [comment]
> > There are some issues with your changes regarding `@property` in NSAppearance. Please correct these before merging.
> 
> @gcasa I see that I've forgotten to implement these properties. I might have accidentally removed that when I rebased on the current master. Are there any issues in the header file?

Oh, these properties are not implemented because they are in a protocol.

**Comment by fredkiefer** [comment]
On the gcc pipeline, we get this warning:
```
NSView.m:5260:1: warning: incomplete implementation of class ‘NSView’
 5260 | @end
      | ^
NSView.m:5260:1: warning: method definition for ‘-setAppearance:’ not found
NSView.m:5260:1: warning: method definition for ‘-appearance’ not found
NSView.m:5260:1: warning: method definition for ‘-effectiveAppearance’ not found
NSView.m:5260:1: warning: class ‘NSView’ does not fully implement the ‘NSAppearanceCustomization’ protocol
```

**Comment by ethanc8** [comment]
> On the gcc pipeline, we get this warning:
> 
> ```
> NSView.m:5260:1: warning: incomplete implementation of class ‘NSView’
>  5260 | @end
>       | ^
> NSView.m:5260:1: warning: method definition for ‘-setAppearance:’ not found
> NSView.m:5260:1: warning: method definition for ‘-appearance’ not found
> NSView.m:5260:1: warning: method definition for ‘-effectiveAppearance’ not found
> NSView.m:5260:1: warning: class ‘NSView’ does not fully implement the ‘NSAppearanceCustomization’ protocol
> ```
> 
> For clang we get:
> 
> ```
>  Compiling file NSView.m ...
> NSView.m:132:17: warning: auto property synthesis will not synthesize property 'appearance' declared in protocol 'NSAppearanceCustomization' [-Wobjc-protocol-property-synthesis]
> @implementation NSView
>                 ^
> ../Headers/AppKit/NSAppearance.h:89:33: note: property declared here
> @property(retain) NSAppearance* appearance;
>                                 ^
> NSView.m:5260:1: note: add a '@synthesize' directive
> @end
> ^
> NSView.m:132:17: warning: auto property synthesis will not synthesize property 'effectiveAppearance' declared in protocol 'NSAppearanceCustomization' [-Wobjc-protocol-property-synthesis]
> @implementation NSView
>                 ^
> ../Headers/AppKit/NSAppearance.h:93:43: note: property declared here
> @property(readonly, retain) NSAppearance* effectiveAppearance;
>                                           ^
> NSView.m:5260:1: note: add a '@synthesize' directive
> @end
> ^
> 2 warnings generated.
> ```

@fredkiefer I believe [9202dfe](https://github.com/gnustep/libs-gui/pull/240/commits/9202dfeece79b2c2904967e8438bf8601798e8bd) solves this, but I'm not sure if my implementation of `-effectiveAppearance` is correct. Also, is it correct to use `ASSIGN` or `ASSIGNCOPY` in `-setAppearance:`?

**Comment by gcasa** [comment]
Given @fredkiefer 's approval and that CI is passing I will go ahead and merge this.  @ethanc8 for future reference, when you get approval it is okay for you to go ahead and do the merge so long as CI is passing as well.

**Pull request merged**
